src/main/java/me/totalfreedom/totalfreedommod/command/Command_gtfo.java

```diff
@@ -42,18 +43,34 @@ public boolean run(CommandSender sender, Player playerSender, Command cmd, Strin
 
         FUtil.bcastMsg(player.getName() + " has been a VERY naughty, naughty boy.", ChatColor.RED);
 
-        // Undo WorldEdits
-        try
+        //checks if there is CoreProtect loaded and installed , if not it skips the rollback and uses coreprotect directly
+        if (!getServer().getPluginManager().isPluginEnabled("CoreProtect"))
         {
-            plugin.web.undo(player, 15);
+            // Undo WorldEdits
+            try
+            {
+                plugin.web.undo(player, 15);
+            }
+            catch (NoClassDefFoundError ex)
+            {
+            }
+
+            // Rollback
+            plugin.rb.rollback(player.getName());
+
         }
-        catch (NoClassDefFoundError ex)
+
+        if (getServer().getPluginManager().isPluginEnabled("CoreProtect"))
         {
+            try
+            {
+                server.dispatchCommand(sender, "co rollback t:1d r:#global u:" + player.getName());
```

Comment:
We should not be using a dispatch command. Thats generally considered a big no-no